test/models/user_test.rb

```diff
@@ -172,6 +172,24 @@ class UserTest < ActiveSupport::TestCase
       should_not allow_value("012345678901234567890").for(:twitter_username)
     end
 
+    context "homepage url" do
+      should allow_value("https://www.mywebsite.com").for(:homepage_url)
+      should allow_value("http://www.mywebsite.com").for(:homepage_url)
+      should_not allow_value("<html><body>hi</body><html>").for(:homepage_url)
+      should_not allow_value("javascript:alert('hello');").for(:homepage_url)
+      should_not allow_value("file:///etc/passwd").for(:homepage_url)
+      should_not allow_value("file://C:/Windows/System32/cmd.exe").for(:homepage_url)
+      should_not allow_value("data:text/html,<script>alert('XSS')</script>").for(:homepage_url)
+      should_not allow_value("data:text/html;base64,SGVsbG8sIFdvcmxkIQ==").for(:homepage_url)
+      should_not allow_value("data:text/html,<img src='x' onerror='alert(1)'>").for(:homepage_url)
+      should_not allow_value("data:text/html,<iframe src='javascript:alert(1)'></iframe>").for(:homepage_url)
+      # should_not allow_value("http://www.site.site?#redirect=www.fake-target.site").for(:homepage_url)
+      # should_not allow_value("http://www.site.com/redirect?url=http://www.malicious-site.com").for(:homepage_url)
```

Comment:
I am guessing this is a yes, but I commented this out because these tests will fail, and I wanted to clarify whether these are edge cases we want to cover.